### PR TITLE
CASSANDRA-19704 UnsupportedOperationException when reducing scope for LCS compactions

### PR DESCRIPTION
**Pull request by maedhroz**
patch by Caleb Rackliffe; reviewed by ? for CASSANDRA-19704

### CODE REVIEW

File: src/java/org/apache/cassandra/db/compaction/LeveledCompactionTask.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public CompactionAwareWriter getCompactionAwareWriter(ColumnFamilyStore cfs,
     @Override
     protected boolean partialCompactionsAcceptable()
     {
-        throw new UnsupportedOperationException("This is now handled in reduceScopeForLimitedSpace");
+        return level == 0;
```
Comment (jasonstack) [inline src/java/org/apache/cassandra/db/compaction/LeveledCompactionTask.java:59]: `reduceScopeForLimitedSpace` allows to remove sstable on level 0 for task with `level<=1`.

should we change it to `level <= 1`?

Comment (maedhroz) [reply src/java/org/apache/cassandra/db/compaction/LeveledCompactionTask.java:59]: I think you're right. We could potentially not compact fully expired SSTables if we don't change to `level <= 1` from what I'm reading.

CC @krummas

Comment (maedhroz) [reply src/java/org/apache/cassandra/db/compaction/LeveledCompactionTask.java:59]: Already fixed in the trunk patch, but I might also add some inline commentary: https://github.com/apache/cassandra/pull/3388

Comment (maedhroz) [reply src/java/org/apache/cassandra/db/compaction/LeveledCompactionTask.java:59]: Will fix on commit for 4.0, 4.1, and 5.0

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/d54646e40957ffc18f651753941fac3068b79f8e

**Pull request closed without merging**